src/cesiumutils.js

```diff
@@ -218,3 +218,26 @@ export function cartesianToDegrees(cartesian) {
     cartographic.height
   ];
 }
+
+/**
+ * Extend kml for export with entities properties
+ * @param {string} kml - kml for export
+ * @param {Array<Entity>} entities - list of entities for export
+ * @return {string}
+ */
+export function extendKmlWithProperties(kml, entities) {
+  entities.values.forEach(ent => {
+    let kmlProperties = '<ExtendedData>';
+    ent.properties.propertyNames.forEach(prop => {
+      let value = ent.properties[prop] ? ent.properties[prop].getValue() : undefined;
+      if (value || typeof value === 'number') {
+        value = typeof value === 'object' ? JSON.stringify(value) : value;
```

Comment:
I think this is a bit convoluted.
What about something like?
```js
 const valueType = typeof value;
if (valueType === 'number') {
  kmlProperties += `<Data name="${prop}"><value>${value}</value></Data>`;
} else if (valueType === 'object') {
  kmlProperties += `<Data name="${prop}"><value>${JSON.stringify(value)}</value></Data>`;
}
```

And what about string values or arrays?

Reply:
we don't need to check for 'number', it was only for '0' value. Arrays have type 'object'.

I update the condition to make it more clear